### PR TITLE
Hookify more components

### PR DESCRIPTION
**Pull request by BPScott**
As with other hookify PRs this is probably best viewed with supresed whitespace.

### WHY are these changes introduced?

More work towards #1995.

### WHAT is this pull request doing?

Hookifies RangeSlider, ScrollLock and Topbar's SearchField

### How to 🎩

- Confirm Single / Dual rangeslider examples work
- Confirm Topbar SearchField works, by checking the Frame example (The TopBar example seems to be buggy losing content on blur on master)
- Confirm ScrollLock works by checking modal overlays stop you scrolling the page within web. ScrollLock doesn't work within storybook even on master for whatever reason.

### CODE REVIEW

File: src/components/TopBar/components/SearchField/tests/SearchField.test.tsx [hunk 1]
```diff
@@ -2,10 +2,7 @@ import React from 'react';
 import {CircleCancelMinor} from '@shopify/polaris-icons';
 import {ReactWrapper} from 'enzyme';
 import {mountWithAppProvider} from 'test-utilities/legacy';
-// We've got to export a named SearchField otherwise other stuff breaks
```
Comment (AndrewMusgrave) [inline src/components/TopBar/components/SearchField/tests/SearchField.test.tsx]: Just curious what this was breaking?

File: src/components/ScrollLock/ScrollLock.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import {useEffect} from 'react';
+import {useScrollLockManager} from '../../utilities/scroll-lock-manager';
+import './ScrollLock.scss';
+
+interface ScrollLockProps {}
+
+// Even though this has no args, reference ScrollLockProps so the prop explorer
```
Comment (AndrewMusgrave) [inline src/components/ScrollLock/ScrollLock.ts]: Will this show up in the style guide without a `Props` interface?

File: src/components/ScrollLock/ScrollLock.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import {useEffect} from 'react';
+import {useScrollLockManager} from '../../utilities/scroll-lock-manager';
+import './ScrollLock.scss';
+
+interface ScrollLockProps {}
```
Comment (AndrewMusgrave) [inline src/components/ScrollLock/ScrollLock.ts]:
```suggestion
interface Props {}
```

File: src/components/ScrollLock/ScrollLock.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+// Even though this has no args, reference ScrollLockProps so the prop explorer
+// in the styleguide works without warnings about unfound props
+export function ScrollLock(_: ScrollLockProps) {
```
Comment (AndrewMusgrave) [inline src/components/ScrollLock/ScrollLock.ts]:
```suggestion
export function ScrollLock(_: Props) {
```

File: src/components/ScrollLock/ScrollLock.ts (hunk 2 above)
Comment (AndrewMusgrave) [reply src/components/ScrollLock/ScrollLock.ts]: On second thought, looks like we don't have a readme and this is just an internal component.

Comment (BPScott) [reply src/components/ScrollLock/ScrollLock.ts]: Per my proposal in #1959 we now name Props after the component - see ScrollLockProps above this.

The styleguide was updated to handle this in https://github.com/Shopify/polaris-styleguide/pull/2932

File: src/components/TopBar/components/SearchField/tests/SearchField.test.tsx (hunk 1 above)
Comment (BPScott) [reply src/components/TopBar/components/SearchField/tests/SearchField.test.tsx]: It complained about being able to generate the types for TopBar.SearchField - or rather it was ok generating them, but trying to use it in consuming projects caused complaints. exporting SearchField in addition to the default component seemed to make it happy.

I don't understand the reasoning for it, but I seemed like it was a due to a combo of a trying to use a withAppProvider wrapped component as an exposed subcomponent

Comment (BPScott) [reply src/components/TopBar/components/SearchField/tests/SearchField.test.tsx]: It complained about being able to generate the types for TopBar.SearchField - or rather it was ok generating them, but trying to use it in consuming projects caused complaints. exporting SearchField in addition to the default component seemed to make it happy.

I don't understand the reasoning for it, but I seemed like it was a due to a combo of a trying to use a withAppProvider wrapped component as an exposed subcomponent

File: src/components/TopBar/components/SearchField/SearchField.tsx [hunk 5]
```diff
@@ -33,148 +28,100 @@ export interface SearchFieldProps {
...
-  private handleChange = ({
-    currentTarget,
-  }: React.ChangeEvent<HTMLInputElement>) => {
-    const {onChange} = this.props;
-    onChange(currentTarget.value);
-  };
+  }, [onCancel, onChange]);
+
+  useEffect(() => {
+    if (input.current) {
```
[the remark this answers is not in the document]
Comment (BPScott) [reply src/components/TopBar/components/SearchField/SearchField.tsx]: good call

**Pull request merged**